### ISSUE TITLE
idevice_id --list is returning iPhones that are not attached via USB

### ISSUE THREAD

**Issue by jmoody** [libimobiledevice/libimobiledevice#152]
I am not sure if this is expected behavior or not.

I have rspec tests that target physical devices.  Part of the test setup is to re-install an .ipa on the (test) target devices.  

Prior to Xcode 6, I asked `instruments` for the list of connected devices.

```
$ xcrun instruments -s devices
```

Starting with Xcode 6, iOS 8 devices on the local network appear as available for UIAutomation testing. However, ideviceinstaller cannot install an .ipa over the network.

Until the latest version of libimobiledevice, I have been able to find only those devices that are connected via USB using `idevice_id --list`.

After updating to 1.1.7, I started seeing devices that are not attached via USB appearing in the output of `idevice_id --list`.
### Configuration
- MacOS 10.10.1
- Xcode 6.1.1
- libimobiledevice toolchain version 1.1.7

The offending device is iOS 8.1.1.
### Reproduce
1. Turn on an iOS 8.1.1 device, but don't connect it via USB.
2. `$ idevice_id --list`
### Expected

The device UDID to _not_ appear in the list of devices reported by `idevice_id --list`.
### Found

The device UDID appears.
### Notes

Occasionally, the iOS 8.1.1 device appears 2x in the output:

```
$ idevice_id -l
43be < snip > 91124
89b5 < snip > 0ab7b
43be < snip > 91124
```

**Comment by jmoody** [comment]
> Try to disable your Wifi and see if the devices still show up. ;)

The device appears only once after turning off the WiFi.

This is causing problems with `ideviceinstaller`.  If the device is list 2x, ideviceinstaller cannot install.

**Comment by jmoody** [comment]
>> Try to disable your Wifi and see if the devices still show up. ;)

> The device appears only once after turning off the WiFi.

Spoke too soon.  The device appears even if it is not connected via WiFi

**Comment by FunkyM** [comment]
WiFi communication is unstable and Apple's usbmuxd might have cached an older WiFi connected device to overcome connection drops. This is probably the reason why you still see some devices despite turning off WiFi.

As you can imagine, it doesn't make sense that your device would still be working when neither attached by USB or WiFi.

Anyways, this is an unhandled feature to be fixed in libusbmuxd which needs to learn to skip devices with the "Network" connection type property.

Thus confirming this is a duplicate of libimobiledevice/libusbmuxd#22 (see related ticket for a temporary workaround, too).

We'll probably fix it properly soon in libusbmuxd without implementing WiFi support yet.

**Comment by cbowns** [comment]
I personally love libimobiledevice's ability to connect to devices wirelessly: I use it to debug device sleep/wake cycles, and to monitor a device's behaviour when not attached to power.

**Comment by marloncruz** [comment]
not all wants connection through usb. maybe a filter parameter would be a better solution?

**Comment by chachi** [comment]
Has there been any progress on this front? It can make CI jobs unreliable if other devices connect and disconnect wirelessly and automatically.

**Comment by dnsBlah** [comment]
Please make a argument to allow/disallow to view wireless devices

**Comment by felyppers** [comment]
Hey guys, I had this same issue this week and solved by disabling the wifi service in the MAC.

In your MAC, use the command below to list your network services
networksetup -listallnetworkservices

Write down the service name similar to iPhone USB and run:
networksetup -setnetworkserviceenabled SERVICE_NAME off

After that, restart your mobile and it should work, but I agree that should be a flag to ignore Wi-Fi connections, similar to "ios-deploy --no-wifi" flag.

**Comment by harlentan** [comment]
> Hey guys, I had this same issue this week and solved by disabling the wifi service in the MAC.
> 
> In your MAC, use the command below to list your network services
> networksetup -listallnetworkservices
> 
> Write down the service name similar to iPhone USB and run:
> networksetup -setnetworkserviceenabled SERVICE_NAME off
> 
> After that, restart your mobile and it should work, but I agree that should be a flag to ignore Wi-Fi connections, similar to "ios-deploy --no-wifi" flag.

I can also see duplicate device udid with idevice_id by following your steps.


1. Disable the iTunes wifi sync
2. Disable the iTunes auto sync
3. Disable wifi of my Mac
4. restart the iPhone and Mac.

After all of that, But failed. I can still see duplicate device udid with idevice_id

**Comment by rikan** [comment]
Hi guys, I have the same issue. I found that if the mac and iphone were in the same AP(access point) network, the issue happened. So I connected iphone to the other AP,  the issue was resolved. Hope this is helpful.

**Comment by rbucker** [comment]
In my case my iPhone was connect via USB and WiFi. When I disabled the Wifi and waited a few minutes the second entry was removed. The thing is.... what does it mean to have 2 and does that effect anything downstream? Would be nice if someone in the triage team would make a decision.
